scipy/stats/tests/test_stats.py

```diff
@@ -538,6 +538,98 @@ def test_gh3014(self):
         odds, pvalue = stats.fisher_exact([[1, 2], [9, 84419233]])
 
 
+class TestBarnardExact(object):
+    """Some tests to show that barnard_exact() works correctly.
+    """
+
+    def test_basic(self):
```

Comment:
This is good; in the longer term you may want to learn about `@pytest.mark.parametrize` which is particularly useful when testing a function on many values. Here is 1 of the assert fails the other will not run, while with `@pytest.mark.parametrize` it would run the other and tell you which one(s) are failing. 

Though pytest work with `fixtures` that are pretty complex things to grasp sometimes, and can make the code less readable.